tutorials/07-resource-compose/README.md

```diff
@@ -0,0 +1,358 @@
+        pub var sauceType: String
+        pub var spiciness: String
+        pub var toppings: [String]
+
+        /// Initializes the fields to the given arguments
+        init(pizzaName: String, dough: String, timeToBake: UInt, sauceType: String, spiciness: String, toppings: [String]) {
+            self.pizzaName = pizzaName
+            self.dough = dough
+            self.timeToBake = timeToBake
+            self.sauceType = sauceType
+            self.spiciness = spiciness
+            self.toppings = toppings
+        }
+    }
+
+    /// init assigns the contract paths
+    init() {
+        self.PizzaStoragePath = /storage/CadenceResourceComposeTutorialPizzaStoragePath
+        self.PizzaAddToppingPrivatePath = /private/CadenceResourceComposeTutorialPizzaPrivatePath
+        self.PizzaShowOrderPublicPath = /public/CadenceResourceComposeTutorialPizzaPrivatePath
+    }
+}
+```
+
+Some important things to notice in the BestPizzaPlace contract:
+
+1. The `destroy()` function of the Pizza resource calls `destroy()` on all contained resources
+2. The `Order` struct which will be used to display the Pizza ingredients
+3. We're creating a public and a private capability path, the latter can only be accessed by the cook who created the pizza with the first transaction
+
+But let's go on with the next step.
+
+## Create a customer account
+
+```console
+flow accounts create
+```
+
+In this tutorial, the name 'customer' is used.
+When asked, choose the local blockchain.
+
+## Prepare the Pizza
+
+This will create a Pizza with Sauce and store it in the emulator-account.
+Also, the aforementioned public and private capability paths are published.
+This is the content of the transaction:
+
+```cadence:title=tx01_PreparePizza.cdc
+import BestPizzaPlace from 0xf8d6e0586b0a20c7
+
+// This transaction creates a pizza resource and moves it to the account storage.
+// It also creates a private capability that allows to add toppings and a public capability
+// which allows to check the order later.
+transaction {
+    let account: AuthAccount
+
+    prepare(acct: AuthAccount) {
+        let dough <- BestPizzaPlace.createDough(grain: BestPizzaPlace.Grain.wheat, timeToBake: 20)
+        let sauce <- BestPizzaPlace.createSauce(name: "Tomato sauce", spiciness: BestPizzaPlace.Spiciness.hot)
+
+        let pizza <- BestPizzaPlace.createPizza(name: "Bufalina", dough: <-dough, sauce: <-sauce)
+
+        // Store the pizza
+        acct.save<@BestPizzaPlace.Pizza>(<-pizza, to: BestPizzaPlace.PizzaStoragePath)
+        // Link capability references
+        acct.link<&AnyResource{BestPizzaPlace.AddTopping}>(BestPizzaPlace.PizzaAddToppingPrivatePath, target: BestPizzaPlace.PizzaStoragePath)
+        acct.link<&AnyResource{BestPizzaPlace.ShowOrder}>(BestPizzaPlace.PizzaShowOrderPublicPath, target: BestPizzaPlace.PizzaStoragePath)
+
+        self.account = acct
+        log("Pizza will be prepared!")
+    }
+
+    post {
+        // TODO: results in failure
```

Comment:
Why is this todo here? If we plan to keep it maybe need to give more context about why

Reply:
I'm getting a weird error from the emulator. I'm going to remove this 'post' block for now and create an issue for the error.